### DISCUSSION TITLE
`log.exchange.formatter = json` has no effect

### DISCUSSION THREAD

**Discussion by Kritnich** [rabbitmq/rabbitmq-server#12810] [Ideas] [answered]
### Describe the bug

Trying to export logs to the system exchange as structured JSON logs with `log.exchange.formatter = json` does not work. The log messages remain in plaintext.

### Reproduction steps

1. Enable log exchange and JSON formatting:
```
log.exchange = true
log.exchange.formatter = json
```
2. Observe relevant queue for log messages

### Expected behavior

Log messages in the queue are JSON-formatted according to the default field map.

### Additional context

The affected instances are running RabbitMQ 3.13.4, Erlang 26.2.5 in a cluster of 3 or 5 nodes.

**Comment by michaelklishin** [comment]
@Kritnich RabbitMQ 3.13.x is [out of community support](https://www.rabbitmq.com/release-information).

JSON formatting does have an effect. I have revisited it thoroughly a few months ago when the [relevant section of the docs](https://www.rabbitmq.com/docs/logging#json) was last updated.
It works perfectly fine with files and standard output streams.

The formatter may or may not be used when the log target is an exchange and not a file or the standard streams. The docs [do not promise that it is a supported combination](https://www.rabbitmq.com/docs/logging#log-exchange).

Just because two features exist does not mean that they are supposed to be combined.

**Comment by michaelklishin** [comment, marked as answer]
If someone wants to contribute a change that would apply JSON formatting to the messages published to `amq.rabbitmq.log`, they are welcome to do it for `4.x`. There won't be any more `3.13.x` open source releases that are not high severity security patches.

This is a combination of two relatively rarely used features. Both work as documented independently.
If someone needs them both, that person can look into contributing it. The core team focuses primarily on issues that affect the majority of users.
